### PR TITLE
docker: remove CATKIN_WS, define ROS_UNDERLAY

### PR DESCRIPTION
**Pull request by rhaschke**
To enable https://github.com/ros-planning/moveit_ci/pull/62, define `ROS_UNDERLAY` in the source container. The `CATKIN_WS` environment variable is not used anymore by `moveit_ci` and is thus removed.

### CODE REVIEW

File: .docker/ci/Dockerfile [hunk 1]
```diff
@@ -6,9 +6,8 @@ MAINTAINER Dave Coleman dave@picknik.ai
 
 ENV TERM xterm
 
-# Setup catkin workspace
-ENV CATKIN_WS=/root/ws_moveit
```
Comment (mlautman) [inline .docker/ci/Dockerfile]: I don't like how `/root/ws_moveit` is now spread across multiple files. Wouldn't it be better to set

```dockerfile
ENV CATKIN_WS=/root/ws_moveit
ENV ROS_UNDERLAY=$CATKIN_WS/install
```

File: .docker/source/Dockerfile [hunk 2]
```diff
@@ -4,12 +4,12 @@
 FROM moveit/moveit:master-ci-shadow-fixed
 MAINTAINER Dave Coleman dave@picknik.ai
 
+ENV ROS_UNDERLAY /root/ws_moveit/install
+WORKDIR $ROS_UNDERLAY/../src
```
Comment (mlautman) [inline .docker/source/Dockerfile]: See point above about restoring `CATKIN_WS` as a variable. That way we can set `WORKDIR $CATKIN_WS/src` which I think is quite a bit cleaner

File: .docker/ci/Dockerfile (hunk 1 above)
Comment (rhaschke) [reply .docker/ci/Dockerfile]: The `CATKIN_WS` folder created/used in the `ci` and `ci-shadow-fixed` containers is only temporary: It's deleted after installing dependencies via rosdep. This folder could go anywhere.
Only the `source` container actually creates a workspace. Hence, it is sufficient to create it there.

Comment (rhaschke) [reply .docker/ci/Dockerfile]: @mlautman Can we proceed with this PR? I would like to use the `source` docker container for MTC as well.

**Comment by rhaschke** [comment]
Ping @mlautman, @davetcoleman.
Would be nice to get this and https://github.com/ros-planning/moveit_ci/pull/62 merged to progress with Travis testing for
- [ ] https://github.com/ros-planning/moveit_tutorials/pull/317 and
- [ ] https://github.com/ros-planning/moveit_visual_tools/pull/47
- [ ] MTC

**Pull request merged**